src/content/keyAutoAdd/lib/helpers.ts

```diff
@@ -0,0 +1,86 @@
+// cSpell:ignore locationchange
+/* eslint-disable @typescript-eslint/no-empty-object-type */
+
+import { withResolvers } from '@/shared/helpers';
+
+interface WaitForOptions {
+  timeout: number;
+}
+
+interface WaitForElementOptions extends WaitForOptions {
+  root: HTMLElement | HTMLHtmlElement | Document;
+}
+
+export function waitForElement<T extends HTMLElement = HTMLElement>(
```

Comment:
For now, yeah (changed approach, reverse engineering API instead of DOM). But will need it in next PR.  Will remove from this PR maybe and add when needed.